single_tech_samples/synapseanalytics/synapseserverless/README.md

```diff
@@ -0,0 +1,179 @@
+# Synapse Analytics Serverless Pools Best Practices using Delta Lake
+
+## Objectives
+
+This sample aims to demonstrate the use of Synapse Serverless Pools with Delta Lake files.
+A second objective includes to share findings around the automation of view creation on top of the Data Lake.
+And lastly, it demonstrates how to implement CLS (Column Level Security) on the delta tables using the views.
+
+## Pre-requirements
+
+- Azure account
+- Visual Studio Code
+
+### Using Dev Containers
+
+To use a Dev Container, you need to have the following software in addition to the previous pre-requisites:
+
+- Docker
+- Visual Studio Code Remote Development Extension Pack
+
+In order to open the project in a container follow the following steps:
+
+- Open Visual Studio Code and clone the repository.
+- Use the .envtemplate file to create a synapsecontainer.env file with your values
+- Hit Ctlr-Shift-P to open the command palette and type Dev Containers: Open Folder in Container ...
+- When prompted, select the directory \modern-data-warehouse-dataops\single_tech_samples\synapseanalytics\synapseserverless
+- Wait for the container to build, check the logs for more information:
+  
+  ![Container Logs](./images/container.jpg)
+
+### Without using Dev Containers
+
+Clone the repository and follow this prerequisites in order to run the sample:
+
+- Login to Azure az login
+- Install jq
+    ```sudo apt-get install jq```
+- Install Az CLI
+    ```curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash``` (for Ubunto or Debian).
+- Define the following environment variables:
+  - export PROJECT=<PROJECT_NAME>
+  - export AZURE_SUBSCRIPTION_ID=<SUBSCRIPTION_ID>
+  - export DEPLOYMENT_ID=dep1 (or any other string). NOTE: if the deployment fails, you need either to delete the resources created or use a different DEPLOYMENT_ID.
+  - export AZURE_LOCATION=westeurope (or any other Azure location)
+
+NOTE: if you need to check the list of env variables use ```printenv```. If you need to delete an env variable use ```unset <NAME_USED_ON_THE_EXPORT>```
+
+## Running the Sample
+
+Navigate to the directory /modern-data-warehouse-dataops/single_tech_samples/synapseanalytics/synapseserverless and run the following command:
+
+```bash -c ./deploy.sh```
+
+Once the sample is deployed in the subscription of choice, navigate to Synapse Studio and either wait for the trigger to start the Pl_NYCTaxi_0_Main pipeline or you can trigger it manually. This pipeline is reponsible to call P_NYCTaxi_1_Setup and P_NYCTaxi_2_Preparation pipelines is sequence.
```

Comment:
Alternatively, this could be a SQL Script that we ask the user to run once, instead of putting this in a pipeline?

like below:

```sql
USE master

create database db_serverless COLLATE Latin1_General_100_BIN2_UTF8
GO

USE db_serverless

CREATE EXTERNAL DATA SOURCE [ext_ds_datalake2] WITH (LOCATION = N'abfs://datalake@lacest11.dfs.core.windows.net')
GO

CREATE PROC [dbo].[CreateDynamicSQLServerlessView]  @viewName varchar(50), @deltaTableName varchar(50), @externalDataSourceName varchar(50), @viewAliasName varchar(50), @format varchar(30)
AS 
BEGIN
DECLARE @sql_statement varchar(MAX)
SET @sql_statement = N'CREATE OR ALTER VIEW ' + @viewName + ' 
	As SELECT * 
	FROM
		OPENROWSET(
		BULK ''' + @deltaTableName + ''',
		DATA_SOURCE = ''' +  @externalDataSourceName + ''',
		FORMAT = ''' + @format + '''
	)as [' + @viewAliasName + '] 
	'
EXEC (@sql_statement)
END

GO
```

Reply:
Good catch! Seems I had the wrong json describing the pipeline. I do have all the operations being resilient to the second run - I'm adding it now. The reason why I don't ask the user to do a one-time run is because this is also something I want to demonstrate with the sample - how to automate the initial deployment as we don't have Az CLI commands to run the scripts in the pool. We can also discuss tomorrow on our call.